test/common/verifier.go

```diff
@@ -0,0 +1,144 @@
+	nodeName                   string
+	projectName                string
+	shootName                  string
+	shootUID                   string
+	rootPodExecutor            framework.RootPodExecutor
+}
+
+// NewVerifier creates a new Verifier.
+func NewVerifier(log logr.Logger,
+	client kubernetes.Interface,
+	echoServerPodIf clientcorev1.PodInterface,
+	echoServerPodName, projectName, shootName, shootUID string) *Verifier {
+	return &Verifier{
+		log:                        log,
+		client:                     client,
+		rsyslogRelpEchoServerPodIf: echoServerPodIf,
+		rsyslogEchoServerPodName:   echoServerPodName,
+		projectName:                projectName,
+		shootName:                  shootName,
+		shootUID:                   shootUID,
+	}
+}
+
+// SetEchoServerPodIfAndName sets the clientcorev1.PodInterface and the name of the rsyslog-relp-echo-server pod to the Verifier.
+func (v *Verifier) SetEchoServerPodIfAndName(echoServerPodIf clientcorev1.PodInterface, echoServerPodName string) {
+	v.rsyslogRelpEchoServerPodIf = echoServerPodIf
+	v.rsyslogEchoServerPodName = echoServerPodName
+}
+
+// VerifyExtensionForNode verifies whether the shoot-rsyslog-relp extension has properly configured
+// the rsyslog service running on the given node.
+func (v *Verifier) VerifyExtensionForNode(ctx context.Context, nodeName string) {
+	v.setPodExecutor(framework.NewRootPodExecutor(v.log, v.client, &nodeName, "kube-system"))
+	v.setNodeName(nodeName)
+
+	v.verifyThatRsyslogIsActiveAndConfigured(ctx, 5*time.Second, 20*time.Second)
+	v.verifyThatLogsAreSentToEchoServer(ctx, "test-program", "1", "this should get sent to echo server")
+	v.verifyThatLogsAreNotSentToEchoServer(ctx, "other-program", "1", "this should not get sent to echo server")
+	v.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "3", "this should not get sent to echo server")
+
+	v.cleanPodExecutor(ctx)
+	v.setNodeName("")
+}
+
+// VerifyExtensionDisabledForNode verifies whether the configuration done by the shoot-rsyslog-relp extension
+// has been properly cleaned up from the given node after the extension is disabled.
+func (v *Verifier) VerifyExtensionDisabledForNode(ctx context.Context, nodeName string) {
+	v.setPodExecutor(framework.NewRootPodExecutor(v.log, v.client, &nodeName, "kube-system"))
+	v.setNodeName(nodeName)
+
+	v.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "1", "this should not get sent to echo server")
+
+	v.cleanPodExecutor(ctx)
+	v.setNodeName("")
+}
+
+func (v *Verifier) setPodExecutor(rootPodExecutor framework.RootPodExecutor) {
+	v.rootPodExecutor = rootPodExecutor
+}
+
+func (v *Verifier) cleanPodExecutor(ctx context.Context) {
+	ExpectWithOffset(2, v.rootPodExecutor.Clean(ctx)).To(Succeed())
+}
+
+func (v *Verifier) setNodeName(nodeName string) {
+	v.nodeName = nodeName
+}
+
+func (v *Verifier) verifyThatRsyslogIsActiveAndConfigured(ctx context.Context, pollInterval, timeout time.Duration) {
+	EventuallyWithOffset(2, func(g Gomega) {
```

Comment:
```suggestion
	EventuallyWithOffset(2, ctx, func(g Gomega) {
```

So that it fails whatever happens first: the ctx is cancelled or the explicit timeout in `.WithTimeout(timeout)` is exceeded.